src/v/kafka/server/audit_log_manager.h

```diff
@@ -0,0 +1,156 @@
+  : public ss::peering_sharded_service<audit_log_manager> {
+public:
+    audit_log_manager(
+      cluster::controller* controller, kafka::client::configuration&);
+
+    audit_log_manager(const audit_log_manager&) = delete;
+    audit_log_manager& operator=(const audit_log_manager&) = delete;
+    audit_log_manager(audit_log_manager&&) = delete;
+    audit_log_manager& operator=(audit_log_manager&&) = delete;
+    ~audit_log_manager();
+
+    /// Start the underlying kafka client and create the audit log topic
+    /// if necessary
+    ss::future<> start();
+
+    /// Shuts down the internal kafka client and stops all pending bg work
+    ss::future<> stop();
+
+    /// Enqueue an event to be produced onto an audit log partition
+    ///
+    /// Returns: bool representing if the audit msg was successfully moved into
+    /// the queue or not. If unsuccessful this means the audit subsystem cannot
+    /// publish messages. Consumers of this API should react accordingly, i.e.
+    /// return an error to the client.
+    template<InheritsFromOCSFBase T, typename... Args>
+    bool enqueue_audit_event(audit_event_type type, Args&&... args) {
+        if (_as.abort_requested()) {
+            /// Prevent auditing new messages when shutdown starts that way the
+            /// queue may be entirely flushed before shutdown
+            return false;
+        }
+        if (!_auditing_enabled() || !is_audit_event_enabled(type)) {
+            return true;
+        }
+        return do_enqueue_audit_event(
+          std::make_unique<T>(std::forward<Args>(args)...));
+    }
+
+    /// Returns the number of items pending to be written to auditing log
+    ///
+    /// Note does not include records already sent to client
+    size_t pending_events() const { return _queue.size(); };
+
+    /// Returns true if the internal kafka client is allocated
+    ///
+    /// NOTE: Only works on shard_id{0}, use in unit tests
+    bool is_client_enabled() const;
+
+private:
+    ss::future<> drain();
+    ss::future<> pause();
+    ss::future<> resume();
+
+    bool is_audit_event_enabled(audit_event_type) const;
+    bool do_enqueue_audit_event(
+      std::unique_ptr<security::audit::ocsf_base_impl> msg);
+    void set_enabled_events();
+
+private:
+    struct underlying_list {};
+    struct underlying_unordered_map {};
+    using underlying_t = boost::multi_index::multi_index_container<
+      std::unique_ptr<security::audit::ocsf_base_impl>,
+      boost::multi_index::indexed_by<
+        /// Sequenced list of entries
+        boost::multi_index::sequenced<boost::multi_index::tag<underlying_list>>,
+        /// Set of audit_messages using hashed representations as comparator
+        boost::multi_index::hashed_unique<
+          boost::multi_index::tag<underlying_unordered_map>,
+          boost::multi_index::const_mem_fun<
+            security::audit::ocsf_base_impl,
+            size_t,
+            &security::audit::ocsf_base_impl::key>>>>;
```

Comment:
can you add a comment above that describes the data model of this index? what is the underlying storage and does that raise concerns about contiguous memory usage?